### ISSUE TITLE
[BUG] Rotosolve no longer solves simple problem

### ISSUE THREAD

**Issue by cvjjm** [PennyLaneAI/pennylane#2100]
### Expected behavior

Expected behavior is observed until 0.20.0:
```
0 -0.06732617961894123
1 -0.9984927436588705
2 -0.9999990754778825
3 -0.9999999973293606
4 -0.9999999999921346
5 -0.999999999999976
6 -0.9999999999999991
7 -0.999999999999999
8 -0.9999999999999986
9 -0.9999999999999989
```

### Actual behavior

In master the optimizer no longer works:
```
0 -0.03383609759966596
1 -0.03383609759966596
2 -0.03383609759966596
3 -0.03383609759966596
4 -0.03383609759966596
5 -0.03383609759966596
6 -0.03383609759966596
7 -0.03383609759966596
8 -0.03383609759966596
9 -0.03383609759966596
```

### Source code

```shell
import pennylane as qml
import numpy as np

wires = 3
dev = qml.device('default.qubit', wires=wires)

def ansatz(params):
    for wire in range(wires):
        qml.Rot(*params[0][wire], wires=wire)
    qml.CNOT(wires=[0,1])
    for wire in range(wires):
        qml.Rot(*params[1][wire], wires=wire)
    qml.CNOT(wires=[1,2])

@qml.qnode(dev)
def cost(params):
    ansatz(params)
    return qml.expval(qml.PauliY(0) @ qml.PauliX(1) @ qml.PauliZ(2))

opt = qml.RotosolveOptimizer()

params = np.random.rand(2, wires, 3)

for i in range(10):
    print(i, cost(params))
    params = opt.step(cost, params)
```

### System information

```shell
Tested with different version.
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by josh146** [comment]
Thanks @cvjjm! This is a very good catch.

**Comment by dwierichs** [comment]
Hi @cvjjm! `Rotosolve` is trying to mimic the training behaviour of gradient-based optimizers, regarding the UI. Therefore it will only train parameters that have `requires_grad` set to `True`. Looks like we definitely need to up the communication on this one! :)

I'm happy to receive any other feedback you might have on the new Rotosolve implementation (a second generalization was merged in but is going to be released only with PennyLane 0.21)

**Comment by cvjjm** [comment]
That makes sense! Maybe it should then throw an error if it is asked to train a qnode without trainable parameters?
